crypto/src/main/scala/coop/rchain/crypto/util/CertificateHelper.scala

```diff
@@ -110,14 +120,65 @@ object CertificateHelper {
     val cert = new X509CertImpl(info)
     cert.sign(privateKey, algorythm)
 
-    // Update the algorith, and resign.
+    // Update the algorithm, and resign.
     val algorithmId2 = cert.get(X509CertImpl.SIG_ALG).asInstanceOf[AlgorithmId]
     info.set(CertificateAlgorithmId.NAME + "." + CertificateAlgorithmId.ALGORITHM, algorithmId2)
     val cert2 = new X509CertImpl(info)
     cert2.sign(privateKey, algorythm)
     cert2
   }
 
+  def encodeSignatureRStoDER(signatureRS: Array[Byte]): Either[Throwable, Array[Byte]] = {
+    def toASN1Int(bytes: Array[Byte]) = new ASN1Integer(BigIntegers.fromUnsignedByteArray(bytes))
+
+    def convert = {
+      val (r, s) = signatureRS.take(64).splitAt(32)
+      val bos    = new ByteArrayOutputStream(72)
+      val seq    = new DERSequenceGenerator(bos)
+      try {
+        seq.addObject(toASN1Int(r))
+        seq.addObject(toASN1Int(s))
+        seq.close
+        bos.toByteArray
+      } finally {
+        if (seq != null) seq.close()
+        // > Closing a ByteArrayOutputStream has no effect.
+        // https://docs.oracle.com/javase/10/docs/api/java/io/ByteArrayOutputStream.html
+      }
+    }
+
+    if (signatureRS.isEmpty)
+      new Exception("Input array must not be empty.").asLeft
+    else
+      Try(convert).toEither
```

Comment:
`Try[A]` is isomorphic to `Either[Throwable, A]` (well, there are edge cases but for practical purposes, it is). I am not sure why we cannot just stick to `Try[A]` then

Reply:
@marcin-rzeznicki is there an implicit converter between these two types? Without explicit conversion `toEither` compiler complains for the wrong type.

Reply:
No, they're isomorphic by "nature". You could switch to `Try` simply - I think it's less hassle than maintaining `Either`. I don't think this is critical though

Reply:
You mean to change return type to `def encodeSignatureRStoDER(signatureRS: Array[Byte]): Try[Array[Byte]]`?

Reply:
I like the `Try` (except mapping failure :smiley:).